### PR TITLE
bucket => run, remove file commands, no more pipe :(

### PR DESCRIPTION
**Pull request by vanpelt**
@shawnlewis lot's of cleanup.  A few comments inline for the interesting ones.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -4,25 +4,25 @@ This package provides an interactive CLI and python client for [wandb.ai](https:
...
 $ wandb init
-$ ./my_training.py arg1 | wandb my_bucket model.json weights.h5
+$ wandb my_script.py arg1
 ```
 
 To manually push files to the cloud run:
 
 ```console
-$ wandb push my-new-bucket somefile.proto
+$ wandb push latest somefile.proto
```
Comment (vanpelt) [inline docs/index.md]: I pushed https://github.com/wandb/core/commit/e6da6d4c774cf992cc0c996c5916b658ce23b386 which will find the latest run if the id is latest.

File: docs/usage.md [hunk 2]
```diff
@@ -14,22 +14,22 @@ Describe the next run using Markdown
 $ wandb describe
 ```
 
-Checkout the commit associated with the bucket and restore the config variables
+Checkout the commit associated with the run and restore the config variables
 
 ```console
-$ wandb restore
+$ wandb restore run-id
```
Comment (vanpelt) [inline docs/usage.md]: I was thinking, alongside restore we could potentially have `wandb rerun run-id` which would restore and run the same command as used before.  Was also thinking we probably don't want to download the non-wandb run files when restoring...

File: wandb/__init__.py [hunk 3]
```diff
@@ -14,11 +14,6 @@
     __stage_dir__ = '.wandb/'
 else:
     __stage_dir__ = "wandb/"
-if not os.path.exists(__stage_dir__):
-    try:
-        os.mkdir(__stage_dir__)
-    except FileExistsError:
-        pass
```
Comment (vanpelt) [inline wandb/__init__.py]: I'm just removing the auto dir creation, let's chat tomorrow.

File: wandb/api.py [hunk 4]
```diff
@@ -82,7 +82,7 @@ def __init__(self, default_settings=None, load_settings=True):
         self.default_settings = {
             'section': "default",
             'entity': "models",
-            'bucket': "default",
+            'run': "latest",
```
Comment (vanpelt) [inline wandb/api.py]: This allows commands like push and restore to work without specifying a run_id

File: wandb/cli.py [hunk 5]
```diff
@@ -52,41 +52,25 @@ def editor(content='', marker='# Enter a description, markdown is allowed!\n'):
 # returned by api.settings()
 CONTEXT=dict(default_map=api.settings())
 
-class BucketGroup(click.Group):
+class RunGroup(click.Group):
     @display_error
     def get_command(self, ctx, cmd_name):
+        #TODO: check if cmd_name is a file in the current dir and not require `run`?
```
Comment (vanpelt) [inline wandb/cli.py]: I was thinking we could make `wandb` without a sub command run a file pretty easily here.

File: wandb/cli.py [hunk 6]
```diff
@@ -533,10 +470,5 @@ def delete(ctx, keys):
     config.persist()
     ctx.invoke(show, changed=changed, diff=True)
 
-@config.command("run", help="Launch a job")
```
Comment (vanpelt) [inline wandb/cli.py]: `@config.command` is actually a sub command of the `@config` group.  Moved it up in the file and used `@cli.command` which is the top level group.

File: wandb/sync.py [hunk 7]
```diff
@@ -117,7 +117,7 @@ def watch(self, files):
                 ), {"diff.patch": open(__stage_dir__+"diff.patch", "rb")})
             self._observer.start()
             print("Syncing %s" % self.url)
-            # Piped mode TODO: remove this gimic
+            # TODO: put the sub process run logic here?
```
Comment (vanpelt) [inline wandb/sync.py]: I didn't rip out the `source_proc` pipe logic, figured it might be useful for the sub process.

File: docs/index.md (hunk 1 above)
Comment (shawnlewis) [reply docs/index.md]: "latest" is a possible auto-generated run id, so we'll either need to blacklist it from being generated or change run_id generation strategy. I'm going to make an issue about this.

Comment (vanpelt) [reply docs/index.md]: Good call 👍

File: wandb/cli.py [hunk 8]
```diff
@@ -52,41 +52,25 @@ def editor(content='', marker='# Enter a description, markdown is allowed!\n'):
...
 @click.pass_context
 def cli(ctx):
     """Weights & Biases
 
-If no command is specified and input is piped, the source command and it's
-output will be saved to the bucket and the files uploaded when modified.
+If the first argument is a file in the current directory run it.
 
-   ./train.sh arg1 arg2 | wandb imagenet/v2 model.json weights.h5
+   wandb train.py --arg=1
```
Comment (shawnlewis) [inline wandb/cli.py]: This feels wrongish to me at first glance. I'll let it simmer for a bit.

**Pull request merged**